src/test/components/elements/cards/QuestionCard.test.tsx

```diff
@@ -0,0 +1,68 @@
+import { screen, waitFor } from "@testing-library/react";
+import { clickButton, renderTestEnvironment } from "../../../utils";
+import QuestionCard from "../../../../app/components/elements/cards/QuestionCard";
+import { mockRandomQuestions } from "../../../../mocks/data";
+
+describe("QuestionCard", () => {
+  const mockSetExpanded = jest.fn();
+
+  window.HTMLElement.prototype.scrollIntoView = jest.fn();
+
+  const setupTest = async () => {
+    renderTestEnvironment({
+      role: "STUDENT",
+      PageComponent: QuestionCard,
+      componentProps: {
+        setExpanded: mockSetExpanded,
+        questionData: mockRandomQuestions,
+      },
+      initialRouteEntries: ["/"],
+    });
+    await waitFor(() => {
+      expect(screen.queryByText("Loading...")).toBeNull();
+    });
+  };
+
+  it("renders question data, next question button, expand button and a check answer button", async () => {
+    await setupTest();
+    const questionTile = screen.getByTestId("question-tile");
+    const nextQuestion = screen.getByText("Next question");
+    const question = screen.getByText(mockRandomQuestions[0].children[0].value!);
+    const checkAnswerButton = screen.getByRole("button", { name: "Check my answer" });
+    const expandButton = screen.getByRole("button", {
+      name: /expand content/i,
+    });
+    [nextQuestion, questionTile, question, checkAnswerButton, expandButton].forEach((element) => {
+      expect(element).toBeInTheDocument();
+    });
+  });
+
+  it("renders the question tile as minimised by default, and expands to panel when the expand button is clicked", async () => {
```

Comment:
Rather than using `toHaveClass` to test this, can't you use `toBeVisible` instead? It would make it a lot more readable (and less likely to change in the future if the classes are renamed, although this is not as important).

Reply:
`toBeVisible` wouldn't work here; pressing the expand button doesn't change the visibility or switch components, it just resizes some elements - as far as I know the only way to verify that is going to be checking the classes.

Reply:
No worries, it was worth a try :-)